### DISCUSSION TITLE
DNAnalyzer v3.5.0-beta.0

### DISCUSSION THREAD

**Discussion by VerisimilitudeX** [VerisimilitudeX/DNAnalyzer#456] [Announcements]
## Introduction

This release introduces full batch FASTA file support, quantitative insights into disease-associated gene expression sites, annotations for CFTR sequences, and parallelized codon mapping for even faster data processing. Additionally, it addresses several major compatibility bugs on Linux and uses Multi-Task Learning (MTL) to further improve performance by learning multiple machine learning tasks at the same time.

DNAnalyzer currently takes just **11 minutes** to conduct analyses on a full human genome running on an M4 Max MacBook Pro, a 47% reduction on average from `v3.0.0-beta.0`. The end goal for DNAnalyzer is to be able to run on a mobile device in mere seconds.

Of course, this release would not have been possible without contributions from the computational biology research community, and we are grateful to the **43 returning and first-time contributors** for their commits, discussion posts, and issues. Please continue reading for more updates and the full change log.

## Updates to Citations

If you utilize this software for personal, institutional, or company purposes, please cite DNAnalyzer using:

```py
Acharya, P. (2022). Democratizing ML-powered DNA analysis through efficient on-device analysis and interpretive tools. (Version 3.5.0-beta.0) [Computer software]. https://github.com/VerisimilitudeX/DNAnalyzer
```

### Features
- **Genes and analysis parameters** by @i2kmt in [#379](https://github.com/VerisimilitudeX/DNAnalyzer/pull/379)
- **Make a full FASTA implementation** by @Nv7-GitHub in [#391](https://github.com/VerisimilitudeX/DNAnalyzer/pull/391)
- **Add basic FastQ implementation** by @Nv7-GitHub in [#399](https://github.com/VerisimilitudeX/DNAnalyzer/pull/399)
- **Genes and their evolutionary background involved in diseases** by @VerisimilitudeX in [#398](https://github.com/VerisimilitudeX/DNAnalyzer/pull/398)
- **CFTR Sequence** by @Karzinisierung in [#406](https://github.com/VerisimilitudeX/DNAnalyzer/pull/406)
- **CFTR Annotations** by @Karzinisierung in [#408](https://github.com/VerisimilitudeX/DNAnalyzer/pull/408)

### Fixes
- **Fixes Undocumented class found JAVA-D1000** by @VerisimilitudeX in [#392](https://github.com/VerisimilitudeX/DNAnalyzer/pull/392)
- **Fix broken lines of code badge** by @VerisimilitudeX in [#393](https://github.com/VerisimilitudeX/DNAnalyzer/pull/393)
- **Fix RBC error from Grace C** by @VerisimilitudeX in [#418](https://github.com/VerisimilitudeX/DNAnalyzer/pull/418)
- **Fix Deepsource issues for #422** by @ishche in [#424](https://github.com/VerisimilitudeX/DNAnalyzer/pull/424)

### Code Improvements
- **Parallelize build codon map** by @garyglatfelter in [#407](https://github.com/VerisimilitudeX/DNAnalyzer/pull/407)
- **Switch to personal PayPal** by @VerisimilitudeX in [#389](https://github.com/VerisimilitudeX/DNAnalyzer/pull/389)
- **Update goals** by @VerisimilitudeX in [#388](https://github.com/VerisimilitudeX/DNAnalyzer/pull/388)

### Documentation Updates
- **Add Armaan to citation** by @VerisimilitudeX in [#412](https://github.com/VerisimilitudeX/DNAnalyzer/pull/412)
- **Update README.md** by @IcebergSnow in [#414](https://github.com/VerisimilitudeX/DNAnalyzer/pull/414)
- **Fix typos and broken links in contribution guides** by @taylorreiter in [#425](https://github.com/VerisimilitudeX/DNAnalyzer/pull/425)

### Styling and Formatting
- **Style: format code with Autopep8 and Google Java Format** by @deepsource-autofix in [#450](https://github.com/VerisimilitudeX/DNAnalyzer/pull/450) and other PRs.
- **[ImgBot] Optimize images** by @imgbot in [#396](https://github.com/VerisimilitudeX/DNAnalyzer/pull/396)

### Miscellaneous
- **Better Linux instructions for downloading** by @LimesKey in [#390](https://github.com/VerisimilitudeX/DNAnalyzer/pull/390)
- **Create `codeql.yml`** by @VerisimilitudeX in [#404](https://github.com/VerisimilitudeX/DNAnalyzer/pull/404)
- **Intro Spring Boot** by @MartinGallauner in [#427](https://github.com/VerisimilitudeX/DNAnalyzer/pull/427)

---

## New Contributors
- @i2kmt made their first contribution in [#379](https://github.com/VerisimilitudeX/DNAnalyzer/pull/379)
- @Karzinisierung made their first contribution in [#406](https://github.com/VerisimilitudeX/DNAnalyzer/pull/406)
- @IcebergSnow made their first contribution in [#413](https://github.com/VerisimilitudeX/DNAnalyzer/pull/413)
- @taylorreiter made their first contribution in [#425](https://github.com/VerisimilitudeX/DNAnalyzer/pull/425)
- @Mrigankkh made their first contribution in [#431](https://github.com/VerisimilitudeX/DNAnalyzer/pull/431)

---

**Full Changelog**: [v3.0.0-beta.0...v3.5.0-beta.0](https://github.com/VerisimilitudeX/DNAnalyzer/compare/v3.0.0-beta.0...v3.5.0-beta.0)

<hr /><em>This discussion was created from the release <a href='https://github.com/VerisimilitudeX/DNAnalyzer/releases/tag/v3.5.0-beta.o'>DNAnalyzer v3.5.0-beta.0</a>.</em>

**Comment by IcebergSnow** [comment]
Awesome! This is perfect for getting metrics quickly for my gene expression project!

**Reply by VerisimilitudeX** [reply]
Glad to hear, @IcebergSnow.
